src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -21,10 +21,15 @@ public sealed class CSharpParseOptions : ParseOptions, IEquatable<CSharpParseOpt
         private ImmutableDictionary<string, string> _features;
 
         /// <summary>
-        /// Gets the language version.
+        /// Gets the effective language version.
         /// </summary>
         public LanguageVersion LanguageVersion { get; private set; }
 
+        /// <summary>
+        /// Gets the specified language version.
+        /// </summary>
+        public LanguageVersion SpecifiedLanguageVersion { get; private set; }
```

Comment:
It is computed when the language version is specified, and then stored. That is to avoid it being computed over and over again as we check the syntax in the parser.